src/typeorm-aurora-data-api-driver.ts

```diff
@@ -1,95 +1,113 @@
 // @ts-ignore
 import createDataApiClient from 'data-api-client'
 
-const quoteCharacters = ['\'', '"']
-
-export default class DataApiDriver {
-  public static transformQueryAndParameters(query: string, parameters?: any[]): any {
-    let numberOfParametersInQueryString = 0
-
-    let newQueryString = ''
-    let currentQuote = null
-
-    for (let i = 0; i < query.length; i += 1) {
-      const currentCharacter = query[i]
-      const currentCharacterEscaped = i !== 0 && query[i - 1] === '\\'
-
-      if (currentCharacter === '?' && !currentQuote) {
+const transformQuery = (query: string, parameters?: any[]): [string, number] => {
+  const quoteCharacters = ['\'', '"']
+
+  let newQueryString = ''
+  let currentQuote = null
+  let numberOfParametersInQueryString = 0
+  let extendedParameterCount = 0
+
+  for (let i = 0; i < query.length; i += 1) {
+    const currentCharacter = query[i]
+    const currentCharacterEscaped = i !== 0 && query[i - 1] === '\\'
+
+    if (currentCharacter === '?' && !currentQuote) {
+      const parameter = parameters![numberOfParametersInQueryString]
+      if (Array.isArray(parameter)) {
+        parameter.forEach((_, index) => {
+          const isLastParameter = index === parameter.length - 1
+          const paramName = `:e_param_${extendedParameterCount}${isLastParameter ? '' : ', '}`
+          extendedParameterCount += 1
+          newQueryString += paramName
+        })
+      } else {
         const paramName = `:param_${numberOfParametersInQueryString}`
-
         numberOfParametersInQueryString += 1
         newQueryString += paramName
-      } else if (quoteCharacters.includes(currentCharacter) && !currentCharacterEscaped) {
-        if (!currentQuote) {
-          currentQuote = currentCharacter
-        } else if (currentQuote === currentCharacter) {
-          currentQuote = null
-        }
-
-        newQueryString += currentCharacter
-      } else {
-        newQueryString += currentCharacter
       }
+
+      continue
```

Comment:
I don't think we should use continue as it makes the code harder to understand

Reply:
Sure I will replace this with an else branch